source/adapters/opencl/device.cpp

```diff
@@ -345,6 +345,23 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
 
     return ReturnValue(URDeviceType);
   }
+  case UR_DEVICE_INFO_DEVICE_ID: {
+    bool Supported = false;
+    CL_RETURN_ON_FAILURE(cl_adapter::checkDeviceExtensions(
+        cl_adapter::cast<cl_device_id>(hDevice), {"cl_khr_pci_bus_info"},
+        Supported));
+
+    if (!Supported) {
+      return UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION;
+    }
+
+    cl_device_pci_bus_info_khr PciInfo = {};
+    CL_RETURN_ON_FAILURE(clGetDeviceInfo(
+        cl_adapter::cast<cl_device_id>(hDevice), CL_DEVICE_PCI_BUS_INFO_KHR,
+        sizeof(PciInfo), &PciInfo, nullptr));
+    return ReturnValue(PciInfo.pci_device);
```

Comment:
I think we could check here for the version of opencl first as I think this query should be only available from version 3.0, I guess we could guard that behind a preprocessor if, not sure if this should work in a preprocessor or not but there is `UR_EXT_DEVICE_INFO_OPENCL_C_VERSION`define in `plugins/unified_runtime/ur/ur.hpp`

Reply:
I think regardless of underlying driver version we don't support building with a CL loader/headers that would be old enough not to have this, if it isn't supported by the device that should get caught by the check for cl_khr_pci_bus_info above

Reply:
ah okay, that make sense. I thought we could as I saw cmake searching for cl version 1.0 and 2.0 before moving to fetching headers from upstream.